src/workspace-data/upload-data/UploadData.test.tsx

```diff
@@ -73,65 +53,125 @@ jest.mock('src/workspaces/common/state/useWorkspaces', () => ({
   }),
 }));
 
-jest.mock('src/libs/feature-previews', () => ({
-  isFeaturePreviewEnabled: jest.fn(),
-}));
+jest.mock('src/components/Dropzone', () => {
+  const actual = jest.requireActual('src/components/Dropzone');
+  return {
+    ...actual,
+    __esModule: true,
+    default: jest.fn(actual.default),
+  };
+});
 
 describe('UploadData Component', () => {
+  const files: File[] = [new File(['entity:sample_id\tname'], 'test1.tsv', { type: 'text/tab-separated-values' })];
+
   beforeEach(() => {
     jest.clearAllMocks();
+
+    asMockedFn(Workspaces).mockReturnValue(
+      partial<WorkspacesAjaxContract>({
+        getTags: jest.fn().mockResolvedValue([]),
+      })
+    );
+
+    asMockedFn(Metrics).mockReturnValue(partial<MetricsContract>({ captureEvent: jest.fn() }));
+
+    asMockedFn(GoogleStorage).mockReturnValue(
+      partial<GoogleStorageContract>({
+        list: jest.fn(async (_googleProject, _bucket, _prefix) =>
+          partial<GCSListObjectsResponse>({
+            items: [partial<GCSItem>({ name: 'test1.tsv', size: '100', updated: '2023-01-01T00:00:00Z' })],
+          })
+        ),
+        listAll: jest.fn(async (_googleProject, _bucket, _prefix) => ({ items: [] })),
+        upload: jest.fn(async () => undefined),
+      })
+    );
+
+    asMockedFn(Dropzone).mockImplementation((props) => {
+      const { onDropAccepted, children } = props;
+      const openUploader = async () => {
+        onDropAccepted && onDropAccepted(files, {} as DropEvent);
+      };
+      return <div>{children(partial<DropzoneState>({ dragging: false, openUploader, ...{} }))}</div>;
+    });
   });
 
   const selectOptionAndGoToNext = async (option: any) => {
     (useRoute as jest.Mock).mockReturnValue({
       query: { ...option },
     });
 
-    await act(async () => {
+    await waitFor(() => {
       renderWithAppContexts(<UploadData />);
     });
 
     fireEvent.click(screen.getByText('Next >'));
   };
 
+  const uploadDataFileToStorage = async () => {
+    const uploadBtn = screen.getByText('Upload');
+
+    await expect(uploadBtn).toBeInTheDocument();
+
+    fireEvent.click(uploadBtn);
+  };
+
   it('Renders data uploader', () => {
+    // Arrange
     renderWithAppContexts(<UploadData />);
+
+    // Act & Assert
```

Comment:
this block of tests look like they should all have the last section just labeled as // Assert, and you can either add Act to the // Arrange comment, or leave it out.

Reply:
Updated: https://github.com/DataBiosphere/terra-ui/pull/5181/commits/e3ef0c7a04fabe0c854833a0389d4b597c36231b